lib/cli/command.ts

```diff
@@ -1,27 +1,56 @@
 import fs from 'fs';
 import grpc, { status } from 'grpc';
+import path from 'path';
 import { Arguments } from 'yargs';
+import Config from '../Config';
 import { XudClient, XudInitClient } from '../proto/xudrpc_grpc_pb';
-import { getDefaultCertPath } from './utils';
 
 /**
  * A generic function to instantiate an XU client.
  * @param argv the command line arguments
  */
-export const loadXudClient = (argv: Arguments<any>) => {
-  const certPath = argv.tlscertpath || getDefaultCertPath();
+export const loadXudClient = async (argv: Arguments<any>) => {
+  const config = new Config();
+  await config.load({
+    xudir: argv.xudir,
+    rpc: {
+      port: argv.rpcport,
+      host: argv.rpchost,
+    },
+  });
+
+  const certPath = argv.tlscertpath || path.join(config.xudir, 'tls.cert');
   const cert = fs.readFileSync(certPath);
   const credentials = grpc.credentials.createSsl(cert);
 
-  return new XudClient(`${argv.rpchost}:${argv.rpcport}`, credentials);
+  // in case port and host args were not set, we update them to the values we
+  // determined by loading the config
+  argv.rpcport = config.rpc.port;
+  argv.rpchost = config.rpc.host;
+
+  return new XudClient(`${argv.rpchost}}:${argv.rpcport}`, credentials);
```

Comment:
Looks like I typoed an extra `}` here after `rpchost` which is why the DNS resolution failures were happening, it was trying to resolve `localhost}` instead of `localhost`. Took me a while to figure out what was going on because I could've sworn it was working for me when I had tested it then I was getting the dns error too out of the blue. Thanks for catching this @erkarl .